src/app/backend-api.service.ts

```diff
@@ -1259,6 +1261,13 @@ export class BackendApiService {
     });
   }
 
+  GetFullTikTokURL(endpoint: string, UpdaterPublicKeyBase58Check: string, TikTokShortVideoID: string): Observable<any> {
+    console.log("over here");
+    return this.jwtPost(endpoint, BackendRoutes.RoutePathGetFullTikTokURL, UpdaterPublicKeyBase58Check, {
+      TikTokShortVideoID,
+    });
```

Comment:
I'd suggest putting the implementation detail of parsing the response here. Look in this file for examples of calling `map()` on a Promise and mapping the response to `FullTikTokURL`

Reply:
I missed this one. let me know if you think it's still cleaner to move over here.

Reply:
I'm in favor of keeping the logic that extracts the video ID from the URL in the video url parser service and not calling it in the backend api service.  Adding it there would introduce a circular dependency